File: SuBilet-frontend/src/services/AirportService.js
import apiClient from '../utils/apiClient';

const API_URL = '/airports';

class AirportService {
    
    getAllAirports() {
        return apiClient.get(API_URL);
    }

    getAirportById(id) {
        return apiClient.get(`${API_URL}/${id}`);
    }

    getAirportByCode(code) {
        return apiClient.get(`${API_URL}/code/${code}`);
    }

    createAirport(airport) {
        return apiClient.post(API_URL, airport);
    }

    updateAirport(id, airport) {
        return apiClient.put(`${API_URL}/${id}`, airport);
    }

    deleteAirport(id) {
        return apiClient.delete(`${API_URL}/${id}`);
    }
}

const airportServiceInstance = new AirportService();
export default airportServiceInstance;

File: SuBilet-backend/src/main/java/com/subilet/backend/service/AirportService.java
package com.subilet.backend.service;

import com.subilet.backend.entity.Airport;
import com.subilet.backend.entity.City;
import com.subilet.backend.exception.ResourceNotFoundException;
import com.subilet.backend.repository.AirportRepository;
import com.subilet.backend.repository.CityRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class AirportService {

    @Autowired
    private AirportRepository airportRepository;

    @Autowired
    private CityRepository cityRepository;

    public List<Airport> getAllAirports() {
        return airportRepository.findAll();
    }

    public Optional<Airport> getAirportById(Integer id) {
        return airportRepository.findById(id);
    }

    public Optional<Airport> getAirportByCode(String code) {
        return airportRepository.findAll().stream()
                .filter(a -> a.getCode().equalsIgnoreCase(code))
                .findFirst();
    }

    public Airport createAirport(Airport airport) {
        // City ilişkisini veritabanından yükle
        if (airport.getCity() != null && airport.getCity().getCityId() != null) {
            City city = cityRepository.findById(airport.getCity().getCityId())
                    .orElseThrow(() -> new ResourceNotFoundException("Şehir bulunamadı"));
            airport.setCity(city);
        }
        return airportRepository.save(airport);
    }

    public Airport updateAirport(Integer id, Airport airportDetails) {
        Airport airport = airportRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Havalimanı bulunamadı: " + id));

        airport.setCode(airportDetails.getCode());
        airport.setName(airportDetails.getName());
        
        // City ilişkisini veritabanından yükle
        if (airportDetails.getCity() != null && airportDetails.getCity().getCityId() != null) {
            City city = cityRepository.findById(airportDetails.getCity().getCityId())
                    .orElseThrow(() -> new ResourceNotFoundException("Şehir bulunamadı"));
            airport.setCity(city);
        }

        return airportRepository.save(airport);
    }

    public void deleteAirport(Integer id) {
        airportRepository.deleteById(id);
    }
}

